File: nvpl_blas/c/zcopy.c
/******************************************************************************
 * Content: 
 *     This example demonstrates use of API as below:
 *     cblas_zcopy
 *
 ******************************************************************************/
#include "example_helper.h"

int main() {
    nvpl_int_t N = 5;
    nvpl_dcomplex_t * X;
    nvpl_dcomplex_t * Y;
    nvpl_int_t incX = 1;
    nvpl_int_t incY = 1;

    printf("\nExample: cblas_zcopy for copying the vector x into the vector y\n\n");
    printf("#### args: n=%" PRId64 ", incx=%" PRId64 ", incy=%" PRId64 "\n", (int64_t)N,
            (int64_t)incX, (int64_t)incY);

    nvpl_int_t len_x = 1 + (N - 1) * labs(incX);
    nvpl_int_t len_y = 1 + (N - 1) * labs(incY);
    // allocate memory
    X = (nvpl_dcomplex_t *)malloc(len_x * sizeof(nvpl_dcomplex_t));
    Y = (nvpl_dcomplex_t *)malloc(len_y * sizeof(nvpl_dcomplex_t));

    // fill data
    fill_zvector(X, N, incX);
    fill_zvector(Y, N, incY);

    // print input data
    print_zvector(X, N, incX, "X");
    print_zvector(Y, N, incY, "Y");

    // call cblas_zcopy
    cblas_zcopy(N, X, incX, Y, incY);

    // print result
    printf("\nThe result of copying the vector x into the vector y:\n");
    print_zvector(X, N, incX, "X");
    print_zvector(Y, N, incY, "Y");

    // release memory
    free(X);
    free(Y);
    return 0;
}

File: nvpl_blas/f77/zcopy.f
!******************************************************************************
! Content:
!     This example demonstrates use of API as below:
!     ZCOPY 
!
!******************************************************************************
      program   ZCOPY_MAIN
*
      implicit none

      include "nvpl_blas_f77_blas.fi"

*       Intrinsic Functions
      intrinsic  abs

      integer         n, incx, incy, i
      parameter       (n=5, incx=1, incy=1)
      integer         xsize, ysize
      parameter       (xsize=1+(n-1)*abs(incx), ysize=1+(n-1)*abs(incy))
      double complex  x(xsize), y(ysize)
      data            (x(i), i=1,xsize)/1.0,-3.7,4.3,0.0,5.0/,
     &                (y(i), i=1,ysize)/ysize*2.0/

*       External Subroutines
      external        print_zvector 
*
*      Executable Statements
*
      print*
      print 99
*
*       Print input data
      print*
      print 100, n, incx, incy
      call print_zvector(x,n,incx,'X')
      call print_zvector(y,n,incy,'Y')
*
*      Call ZCOPY subroutine
      call ZCOPY(n,x,incx,y,incy)
*
      print*
      print 101
      call print_zvector(x,n,incx,'X')
      call print_zvector(y,n,incy,'Y')

      stop
 99   format('Example: SCOPY for copying the vector x into the'
     &       ' vector y')
 100  format('#### args: n=',i1,', incx=',i1,', incy=',i1)
 101  format('The result of copying the vector x into the vector y:')
      end
